### PR TITLE
Describe new ce-knativedispatcherr cloudevent extension attribute

### PR DESCRIPTION
**Pull request by travis-minke-sap**
This PR is associated with the [knative-eventing PR 4760](https://github.com/knative/eventing/pull/4760) which is adding an extension attribute to CloudEvents prior to sending them to the dead letter sink.

## Proposed Changes
- Cleanup KafkaChannel descriptions to reflect the latest implementations.
- Add brief description of new CloudEvent extension attribute `ce-knativedispatcherr` to event-delivery documentation.

### CODE REVIEW

File: docs/eventing/event-delivery.md [hunk 1]
```diff
@@ -90,6 +90,32 @@ spec:
...
+        - "500"
+        - ...any HTTP StatusCode...
+
+- **knativeerrordata**
+    - **Type:** String
+    - **Description:** The HTTP Response **Body** from the final event dispatch
+      attempt.
+    - **Constraints:** Will not be present if the HTTP Response **Body** was
+      empty. Will be truncated to a maximum length of 1024 bytes which could
+      render JSON payloads invalid / incomplete.
```
Comment (slinkydeveloper) [inline docs/eventing/event-delivery.md:113]: Keep it vague the maximum length truncation, so we can eventually modify the length in future and people won't depend on it... Something like:

> The original Response body might be truncated

File: docs/eventing/event-delivery.md [hunk 2]
```diff
@@ -90,6 +90,32 @@ spec:
     backoffDelay: <ISO8601 duration>
 ```
 
+Failed events may, depending on the specific Channel implementation in use, be
+enhanced with extension attributes prior to forwarding to the`deadLetterSink`.
+These extension attributes are as follows...
```
Comment (slinkydeveloper) [inline docs/eventing/event-delivery.md:95]: s/.../:

**Review by slinkydeveloper — COMMENTED** [review]
Small nits, otherwise LGTM

File: docs/eventing/event-delivery.md (hunk 1 above)
Comment (travis-minke-sap) [reply docs/eventing/event-delivery.md:113]: 👍

File: docs/eventing/event-delivery.md (hunk 2 above)
Comment (travis-minke-sap) [reply docs/eventing/event-delivery.md:95]: 👍

**Comment by csantanapr** [comment]
thank you for the contribution

**Pull request merged**